cmd/client_membership.go

```diff
@@ -109,30 +110,46 @@ func runClientMembership(cmd *cobra.Command, args []string) error {
 	fmt.Printf(" ActualVersion: %d\n", membershipResult.ActualVersion)
 	fmt.Printf(" KeyDigest: %x\n\n", membershipResult.KeyDigest)
 
-	if params.Verify {
-		hyperDigest := params.HyperDigest
-		historyDigest := params.HistoryDigest
-		for hyperDigest == "" {
-			hyperDigest = readLine(fmt.Sprintf("Please, provide the hyperDigest for current version [ %d ]: ", membershipResult.CurrentVersion))
-		}
-		if membershipResult.Exists {
-			for historyDigest == "" {
-				historyDigest = readLine(fmt.Sprintf("Please, provide the historyDigest for version [ %d ] : ", params.Version))
+	// Verify
+	if params.Verify || params.AutoVerify {
+		var snapshot *protocol.Snapshot
+
+		if params.Verify {
+			hyperDigest := params.HyperDigest
+			historyDigest := params.HistoryDigest
+			for hyperDigest == "" {
+				hyperDigest = readLine(fmt.Sprintf("Please, provide the hyperDigest for current version [ %d ]: ", membershipResult.CurrentVersion))
+			}
+			if membershipResult.Exists {
+				for historyDigest == "" {
+					historyDigest = readLine(fmt.Sprintf("Please, provide the historyDigest for version [ %d ] : ", params.Version))
+				}
 			}
+			hdBytes, _ := hex.DecodeString(hyperDigest)
+			htdBytes, _ := hex.DecodeString(historyDigest)
+
+			snapshot = &protocol.Snapshot{
+				HistoryDigest: htdBytes,
+				HyperDigest:   hdBytes,
+				Version:       params.Version,
+				EventDigest:   digest}
+
+			fmt.Printf("\nVerifying with Snapshot: \n\n EventDigest: %x\n HyperDigest: %s\n HistoryDigest: %s\n Version: %d\n",
+				digest, hyperDigest, historyDigest, params.Version)
 		}
 
-		hdBytes, _ := hex.DecodeString(hyperDigest)
-		htdBytes, _ := hex.DecodeString(historyDigest)
-		snapshot := &protocol.Snapshot{
-			HistoryDigest: htdBytes,
-			HyperDigest:   hdBytes,
-			Version:       params.Version,
-			EventDigest:   digest}
+		if params.AutoVerify {
+			snapshot = &protocol.Snapshot{
+				HistoryDigest: nil,
+				HyperDigest:   nil,
+				Version:       params.Version,
+				EventDigest:   digest}
 
-		fmt.Printf("\nVerifying with Snapshot: \n\n EventDigest: %x\n HyperDigest: %s\n HistoryDigest: %s\n Version: %d\n",
-			digest, hyperDigest, historyDigest, params.Version)
+			fmt.Printf("\nAuto-Verifying with Snapshot: \n\n EventDigest: %x\n Version: %d\n",
+				digest, params.Version)
+		}
 
-		if client.DigestVerify(membershipResult, snapshot, hasherF) {
+		if client.MembershipVerify(membershipResult, snapshot, hasherF) {
```

Comment:
change snapshot parameter for version and event
membership result should be from balloon instead of protocol